xtrack/madng_interface.py

```diff
@@ -391,11 +515,481 @@ def prepare(self, force=False):
         if init is not None and start is not None and end is not None:
             assert isinstance(init, xt.TwissTable)
             self.X0 = madng_get_init(self.line, at=start)
+        elif init is not None:
+            assert isinstance(init, xt.TwissTable)
+            self.X0 = madng_get_init(self.line, at=xt.START)
+
         self._alredy_prepared = True
 
     def run(self):
         return self.line.madng_twiss(xsuite_tw = False, X0=self.X0, **self.tw_kwargs)
 
+class ActionTwissMadngTPSA(Action):
+    def __init__(self, line, vary_names, targets = [], tw_kwargs={}, twiss_flag=True, sum_rmat_tar=0, **kwargs):
+        self.line = line
+        self.vary_names = vary_names
+        self.targets = targets
+        self.optics_target_locations = None
+        self.optics_target_quantities = None
+        self.tw_kwargs = tw_kwargs
+        self.tw_kwargs.update(kwargs)
+        self.twiss_flag = twiss_flag
+        self._already_prepared = False
+        self.match_rmat = False
+        self.match_opt = False
+        self.sum_rmat_tar = sum_rmat_tar
+        self.rmat_start_end_list = None
+        self.rmat_tags = None
+        self._last_res = None
+        self._needs_zeta_scale = []
+        self._needs_delta_scale = []
+
+    def prepare(self, force=False):
+        '''
```

Comment:
Sorry to be picky, since I know we have this everywhere, but generally docstrings are recommended to use `"""` instead of `'''` ([PEP 257 -- Docstring Conventions](http://www.python.org/dev/peps/pep-0257/)).